actions/team_actions.jsx

```diff
@@ -155,3 +155,12 @@ export async function getTeamMembersForUser(userId, success, error) {
         error({id: err.server_error_id, ...err});
     }
 }
+
+export async function setTeamIcon(teamId, teamIcon, success, error) {
```

Comment:
Instead of creating a wrapper function for this action, can we pass it directly as a prop to the component through it's `index.js`? I know we're doing it this way in a bunch of places but that's actually just for legacy components which haven't been updated yet

For example, it should look similar to this
* https://github.com/mattermost/mattermost-webapp/blob/master/components/add_users_to_team/index.js#L19
* https://github.com/mattermost/mattermost-webapp/blob/master/components/add_users_to_team/add_users_to_team.jsx#L28

Then you can await on the action like you do here with `const {data, error: err} = await this.props.actions.setTeamIcon(teamId, teamIcon);`